backend/tracim_backend/lib/core/mention.py

```diff
@@ -23,30 +22,43 @@
 from tracim_backend.models.event import EntityType
 from tracim_backend.models.event import Event
 from tracim_backend.models.event import OperationType
-from tracim_backend.models.mention import ALL__GROUP_MENTIONS
+from tracim_backend.models.roles import WorkspaceRoles
 from tracim_backend.models.tracim_session import TracimSession
 
 
+class MentionType:
+    USER = 1
+    ROLE = 2
+
+
 class Mention:
     """A mention with its attributes: id and recipient."""
 
-    def __init__(self, recipient: str, id_: str) -> None:
+    def __init__(self, type: MentionType, recipient: int, content_id: int) -> None:
+        self.type = type
         self.recipient = recipient
-        self.id = id_
+        self.content_id = content_id
 
     def __eq__(self, other) -> bool:
         return (
-            isinstance(other, Mention) and other.recipient == self.recipient and other.id == self.id
+            isinstance(other, Mention)
+            and self.type == other.type
+            and self.recipient == other.recipient
+            and self.content_id == other.content_id
         )
 
     def __repr__(self) -> str:
-        return "<Mention(recipient={}, id={})>".format(self.recipient, self.id)
+        return (
+            f"<Mention(type={self.type}, recipient={self.recipient}, content_id={self.content_id})>"
+        )
 
     def __hash__(self) -> int:
-        return hash(self.id)
+        return hash(self.content_id)
```

Comment:
The hash will not be unique when mentioning multiple users in the same content, set it to a mix of type, recipient and content_id.